app/models/course/condition/achievement.rb

```diff
@@ -8,6 +8,16 @@ class Course::Condition::Achievement < ActiveRecord::Base
 
   delegate :title, to: :achievement
 
+  # Checks if the subject has the required achievement.
+  #
+  # @param [Object] subject The subject that the achievement condition is being checked on. The
```

Comment:
If this subject can be duck typed, use #method_name, otherwise use the class name.